src/core/Entry.cpp

```diff
@@ -274,6 +277,63 @@ const EntryAttachments* Entry::attachments() const
     return m_attachments;
 }
 
+bool Entry::hasTotp() const{
+    if (m_attributes->hasKey("otp")) {
+        // Compatibility with "KeeOtp" plugin
+        QRegExp rx("key=(.+)", Qt::CaseInsensitive, QRegExp::RegExp);
+
+        if (rx.exactMatch(m_attributes->value("otp"))) {
+            return true;
+        }
+    } else if (m_attributes->hasKey("TOTP Seed")) {
+        // Compatibility with "Tray TOTP" plugin
+        return true;
+    }
+    return false;
+}
+
+QString Entry::totp() const{
+    if (hasTotp()) {
+        QString seed = "";
+
+        if (m_attributes->hasKey("otp")) {
+            QRegExp rx("key=(.+)", Qt::CaseInsensitive, QRegExp::RegExp);
+            rx.indexIn(m_attributes->value("otp"));
+            seed = rx.cap(1);
+        } else {
+            seed = m_attributes->value("TOTP Seed");
+        }
+
+        const char *secret = qPrintable(seed.toLatin1());
+        char output[20];
```

Comment:
One more thing I just noticed here: isn't this over-allocated? According to the docs, this should have a size of the number of digits + 1 (for the \0 character).

I would also much prefer if you saved the intended number of digits in a constant `numDigits` (`const unsigned`), used the dynamic arrays C99 feature here to allocate the buffer as `numDigits + 1` and then passed this constant as a parameter to `oath_totp_generate()`. That way we avoid possible buffer overflows if someone changes the number of output digits, but doesn't adjust the buffer size or the like.